sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -34,7 +34,8 @@ object NestedColumnAliasing {
     : Option[(Map[ExtractValue, Alias], Map[ExprId, Seq[Alias]])] = plan match {
     case Project(projectList, child)
         if SQLConf.get.nestedSchemaPruningEnabled && canProjectPushThrough(child) =>
-      getAliasSubMap(projectList)
+      val exprCandidatesToPrune = projectList ++ child.expressions
+      getAliasSubMap(exprCandidatesToPrune, child.producedAttributes.toSeq)
```

Comment:
@viirya, just to clarify, you added `producedAttributes` here just to be safe but not related to the current changes (?), for example, if a plan outputs and bypass the attributes from children, we shouldn't prune it like `Expend` case? Seems `Join` and `RepartitionByExpression` have an empty `producedAttributes`.

Reply:
@viirya, just to clarify, you added `producedAttributes` here just to be safe but not related to the current changes (?). Seems `Join` and `RepartitionByExpression` have an empty `producedAttributes`.